### PR TITLE
Feature/best practices

### PR DESCRIPTION
**Pull request by robinsandborg**
Write suggestions for best practices

### CODE REVIEW

File: best-practices/authentication.md
Comment (Gawdfrey) [inline best-practices/authentication.md:1]: Maybe link to [AuthJs](https://authjs.dev/) instead?
You still have to use next-auth for nextjs though

File: best-practices/components/forms.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+### Cons
+- Can be tricky to combine with some component libraries.
+</details>
+
+#### Tips
+<details>
+  <summary>File structure</summary>
+  We recommend moving each field into it's own separate file. This makes it easier to find the field you're looking for, and makes it easier to re-use fields in other forms. It also makes it easier to test the fields.
+
+  In order to keep the details of the field in one place, it can be useful to add `MyField.validationSchema` and `MyField.defaultValue` to the field component.
```
Comment (Gawdfrey) [inline best-practices/components/forms.md:31]: Would probably include a code snippet to show how this is done in practice.

File: best-practices/components/forms.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+</details>
+<details>
+  <summary>useFormConnect</summary>
+  A pattern often used in our projects is to create a `useFormConnect` that wraps the `useController` hook from react-hook-form and gets the field and fieldState.error.message props from the controller and passes them to the component.
```
Comment (Gawdfrey) [inline best-practices/components/forms.md:35]: Either a code snippet or a link to showcase how its done. Easily copy pasteable!

File: best-practices/components/libraries.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+### Pros
+- Well built, high quality components
+- No styling
+- Good documentation
+- Good accessibility
+
+### Cons
+- Cannot be used directly in server components (NextJs app dir)
+- Limited number of components, very oriented towards dashboards
+- Can be difficult to translate styles from the Radix Docs to tailwind.
```
Comment (Gawdfrey) [inline best-practices/components/libraries.md:26]: All(?) radix components have tailwind examples now.

File: best-practices/components/libraries.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+### Pros
+- Well built, high quality components
+- No styling, Tailwind CSS is used in the examples.
+- Good documentation
+- Good accessibility
+
+### Cons
+- Very limited number of components.
+- Cannot be used directly in server components (NextJs app dir)
```
Comment (Gawdfrey) [inline best-practices/components/libraries.md:46]: Would probably nuance this with `These components require javascript, and if you want to utilize these components you need to append them with the "use client" syntax`. This applies to everywhere this sentence is used.

File: best-practices/data-fetching.md
Comment (Gawdfrey) [inline best-practices/data-fetching.md:1]: Should we mention TRPC? We have few use cases on the customer teams, but it is one of the most widely used data fetching libraries.

File: best-practices/project-structure.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Project structure
+The best way to structure your project will be different depending on what kind of project and stack you are using. This document will contain some general tips and recommendations, but you should always use your best judgement and adapt the structure to your specific project.
+
+## General tips
+- Use named exports instead of default exports. This makes it easier to find where a component is imported.
```
Comment (Gawdfrey) [inline best-practices/project-structure.md:5]: Should probably have a small code snippet for this. I have a sense that some might not understand what this means from the getgo

File: best-practices/project-structure.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+- If your project is big, consider using a monorepo. We've had some success with [Turborepo](https://turbo.build/), but there are many other options.
+- Have separate folders for components, hooks, utils, api, pages, etc.
```
Comment (Gawdfrey) [inline best-practices/project-structure.md:7]: We could probably link to one of our templates?
Speaking of templates, we should probably set up a one click next starter in nextjs? @XSlemX @robinsandborg

File: best-practices/seo.md
Comment (Gawdfrey) [inline best-practices/seo.md:1]: Link to Nextjs seo documentation. Should probably have a section on why client side seo does not matter 😂

File: best-practices/testing.md
Comment (Gawdfrey) [inline best-practices/testing.md:1]: Would probably include testwright and vitest as recomendations here.

**Review by Gawdfrey — CHANGES_REQUESTED** [review]
I've left a couple of comments for recommended changes.
Lets discuss and how we will move forward.

File: best-practices/starting-a-new-project.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Starting a new project
+Generally, the recommendation for starting new projects is to [create a new Next.js project](https://nextjs.org/docs/getting-started) using TypeScript.
+
+As of 2023-05-01, we recommend sticking to the tried and true pages structure for Next.js, but we expect this to change soon as the App directory structure becomes more stable and mature.
```
[the remark this answers is not in the document]
Comment (Gawdfrey) [reply best-practices/starting-a-new-project.md:4]: Use a different date format as well

File: best-practices/components/libraries.md (hunk 3 above)
Comment (robinsandborg) [reply best-practices/components/libraries.md:26]: Oh, nice! Did not catch that, neatly hidden away

File: best-practices/starting-a-new-project.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (robinsandborg) [reply best-practices/starting-a-new-project.md:4]: Is it outdated?

File: best-practices/project-structure.md (hunk 6 above)
Comment (Gawdfrey) [reply best-practices/project-structure.md:9]: I was thinking about backstage templates. [This](https://backstage.stacc.dev/create?filters%5Bkind%5D=template&filters%5Buser%5D=all) one for example spins up a fully working flow instance. Backstage can be a nice place to gather them all. But as you mention having one for frontend might not make sense as it is updated so frequently.

File: best-practices/data-fetching.md
Comment (Gawdfrey) [reply best-practices/data-fetching.md:1]: Lets not do that at this time. As mentioned we have rarely used it on our projects so far.

**Review by Gawdfrey — APPROVED** [review]
LGTM! Good first iteration, lets get it out and get some feedback!

**Pull request merged**